### PR TITLE
Several fixes

### PR DESCRIPTION
**Pull request by amiramw**
- environemnt.run return promise and promise doesn't have `on` method
- when no arguments are required for the run there is no need to raise empty prompt
- add support for local generators
- get rid of own implementation of getNpmPaths in favor of reuse of yeoman/environemnt

### CODE REVIEW

**Comment by amiramw** [comment]
fixes #44

File: src/yo/yo.ts [hunk 1]
```diff
@@ -92,57 +82,16 @@ export default class Yeoman {
...
-					} catch (err) {
-						reject(err);
-					}
-
-					rej();
-				});
-			}))).catch(err => {
-				// do nothing because the input is always rejected
-			});
+		return this._env.run(generator, this.done).then(() => {
```
Comment (bfitzpat) [inline src/yo/yo.ts]: Do we no longer need the bower and node dependencies installed here? Or are they installed when the extension itself is added without additional help?

Comment (amiramw) [reply src/yo/yo.ts]: yeoman generator can install bower, npm, yarn, etc. But yeoman environemnt package doesn't expose these events on the run method. Only promise to the end of the generation process is returned.

**Comment by amiramw** [comment]
@bfitzpat for older generators the generator itself is returned from `generator.run` and then the old code will work. For newer generators returning a promise to the end of process the new code will work.

See this commit from 6/2018: https://github.com/yeoman/generator/commit/4f3c6e873b05f4839370d10bade5448efdbe8d77

I will add an if statement and support both as probably some generators didn't have new release in the last year, like this one: https://www.npmjs.com/package/generator-generator which depend on yeoman-generator ^2.0.1.

**Comment by amiramw** [comment]
Done. I also organized tests and added test for my new functionality (listing locally installed generators).

**Comment by bfitzpat** [comment]
> Done. I also organized tests and added test for my new functionality (listing locally installed generators).

Awesome! I'll give this a shot today. I wonder about the test and how it might run on Travis since I don't imagine there are any yeoman generators installed on the virtual machine spun up for the tests. Any thoughts on that?

**Comment by amiramw** [comment]
> Awesome! I'll give this a shot today. I wonder about the test and how it might run on Travis since I don't imagine there are any yeoman generators installed on the virtual machine spun up for the tests. Any thoughts on that?

For my test I installed generator-generator as dev-dependency. You can see it in the package.json.
If we want to test the globally installed generators listing I would try to install them before test run and uninstall after - from the test code.

**Review by bfitzpat — APPROVED** [review]

**Review by bfitzpat — APPROVED** [review]
Everything seems to be working ok locally. Thanks for the submission and the additional tests!

**Pull request merged**